src/main/java/com/shapesecurity/bandolier/es2017/transformations/ImportExportConnector.java

```diff
@@ -215,14 +215,14 @@ private ImportExportConnector() {
 	 * @param moduleAncestorIndex  internal counter, always initialise to 0
 	 * @return an index and state of a module scheduled. the index is used internally, and is to be discarded. success can be determined by checking the module state is `ModuleState.INSTANTIATED`
 	 */
-	private static Pair<Integer, ModuleState> scheduleModule(@Nonnull Module module, @Nonnull LinkedList<Module> schedule, @Nonnull HashSet<Module> scheduleSet, @Nonnull HashMap<Module, LinkedList<Module>> dependingOn, int index, @Nonnull LinkedList<Module> stack, @Nonnull HashSet<Module> stackSet, @Nonnull HashMap<Module, Integer> moduleAncestorIndex) {
-		if (scheduleSet.contains(module)) {
+	private static Pair<Integer, ModuleState> scheduleModule(@Nonnull Module module, @Nonnull LinkedList<Module> schedule, @Nonnull IdentityHashMap scheduleSet, @Nonnull HashMap<Module, LinkedList<Module>> dependingOn, int index, @Nonnull LinkedList<Module> stack, @Nonnull IdentityHashMap stackSet, @Nonnull HashMap<Module, Integer> moduleAncestorIndex) {
```

Comment:
There are still a few `HashMap<Module, Foo>` types in this file. It seems like those would need to be changed too - e.g. if you have two identical modules which are in different paths, they can have different dependencies (and etc). Should those be changed? Either way, can you add a test where the identical modules are in different paths, and therefore have different dependencies? Something like

```js
// ./a.js
import { b as b1 } from './b.js';
import { b as b2 } from './subdir/b.js';
export var result = b1 + b2;

// ./b.js
export { b } from './c.js';

// ./c.js
export var b = 2;

// ./subdir/b.js
export { b } from './c.js';

// ./subdir/c.js
export var b = 3;
``` 

Alternatively, it might be easier to just make a `ModuleWrapper` type that just wraps a `Module` and does not define `hashCode`/`equals`, so there's no risk of accidentally using a `HashMap` instead of an `IdentityHashMap`.

Reply:
I like the `ModuleWrapper` idea better. Though, I need to make it a subclass of `Module` and override the `equals/hashCode` function to make it work because of a number of places where I have change the code this way will be smaller compared to when I implement it as a wrapper.